tracer/src/Datadog.Tracer.Native/cor_profiler.cpp

```diff
@@ -1628,6 +1636,45 @@ HRESULT STDMETHODCALLTYPE CorProfiler::JITCompilationStarted(FunctionID function
             return S_OK;
         }
 
+        // *********************************************************************
+        // Checking if the caller has an explicit static constructor.
+        // If it does, we delay instrumenting this and let the static constructor get instrumented instead.
+        // Bypassing for calls that we explicitly want to instrument (e.g. IIS startup hook)
+        // *********************************************************************
+        if (can_skip_startup_hook_callsite && caller.name != WStr(".cctor"))
+        {
+            mdMethodDef memberDef;
+            hr = metadataImport->FindMethod(caller.type.id, WStr(".cctor"), 0, 0, &memberDef);
+            if (FAILED(hr))
```

Comment:
I remember Kevin saying that `FAILED` and `SUCEEDED` macros are unreliable, should we do `hr != S_OK` then?

Reply:
All our _existing_ usages of `FindMethod()` use `Failed(hr)`, which is why I did here, but it's a good question, @kevingosse? 
![image](https://github.com/user-attachments/assets/8586400b-95ad-4e63-bfb7-0e8cd7b17fa1)

Reply:
Discussed offline

> In the code it doesn't _look_ like it can return `S_FALSE`
> And there are other places where we use it with `if (FAILED(hr))`, so probably fine